src/result/bool32.rs

```diff
@@ -39,18 +44,54 @@ unsafe impl Abi for BOOL {
     type Abi = Self;
 }
 
+impl From<BOOL> for bool {
+    fn from(value: BOOL) -> Self {
+        value == TRUE
```

Comment:
I'm far from a C expert, but shouldn't this be `value != FALSE` or `value > FALSE`?  I thought any non-zero result was true?  Or is this a "windows bool" thing?

Reply:
Well spotted - I thought I'd checked every comparison. Indeed, we should always compare against zero/false.

Reply:
This makes me think we should not derive `PartialEq` and instead normalize when doing equality checks.

Reply:
Fixes here: https://github.com/microsoft/windows-rs/pull/492